grades/api.py

```diff
@@ -0,0 +1,155 @@
+"""
+APIs for the grades app
+"""
+import logging
+from collections import namedtuple
+
+from django.contrib.auth.models import User
+
+from dashboard.api_edx_cache import CachedEdxUserData, CachedEdxDataApi
+from dashboard.models import CachedEnrollment
+from grades.models import (
+    FinalGrade,
+    FinalGradeStatus,
+)
+
+
+log = logging.getLogger(__name__)
+
+UserFinalGrade = namedtuple('UserFinalGrade', ['grade', 'passed'])
+
+
+def _compute_grade_for_fa(user_edx_run_data):
+    """
+    Gets the final grade for an user enrolled in the course run that is part of a financial aid program.
+
+    Args:
+        user_edx_run_data (dashboard.api_edx_cache.UserCachedRunData): the edx cached data for an user in a course run
+
+    Returns:
+        UserFinalGrade: a namedtuple of (float, bool,) representing the final grade
+            of the user in the course run and whether she passed it
+    """
+    run_passed = None
+    grade = None
+    if user_edx_run_data.certificate is not None:
```

Comment:
No, it is because we need to use certificates for old DEDP courses (that are not guaranteed to still have a consistent current grade). A complication is that the certificate is in most cases not verified.